tests/test_modutils.py

```diff
@@ -628,3 +628,18 @@ def test_find_setuptools_pep660_editable_install():
     with unittest.mock.patch.object(sys, "meta_path", new=[_EditableFinder]):
         assert spec.find_spec(["example"])
         assert spec.find_spec(["example", "subpackage"])
+
+
+def test_no_import_done_for_submodule_sharing_std_lib_name() -> None:
+    sys.path.insert(0, resources.find("data"))
+    try:
+        with pytest.raises(ImportError):
+            spec._find_spec_with_path(
```

Comment:
I would have preferred to find a public interface to test, but after almost an hour of fiddling I threw in the towel.